src/core/ext/xds/xds_http_filters.cc

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/ext/xds/xds_http_filters.h"
+
+#include "envoy/extensions/filters/http/router/v3/router.upb.h"
+#include "envoy/extensions/filters/http/router/v3/router.upbdefs.h"
+
+namespace grpc_core {
+
+const char* kXdsHttpRouterFilterConfigName =
+    "envoy.extensions.filters.http.router.v3.Router";
+
+namespace {
+
+class XdsHttpRouterFilter : public XdsHttpFilterImpl {
+ public:
+  void PopulateSymtab(upb_symtab* symtab) const override {
+    envoy_extensions_filters_http_router_v3_Router_getmsgdef(symtab);
+  }
+
+  absl::StatusOr<FilterConfig> GenerateFilterConfig(
+      upb_strview serialized_filter_config, upb_arena* arena) const override {
+    if (envoy_extensions_filters_http_router_v3_Router_parse(
+            serialized_filter_config.data, serialized_filter_config.size,
+            arena) == nullptr) {
+      return absl::InvalidArgumentError("could not parse router filter config");
+    }
+    return FilterConfig{kXdsHttpRouterFilterConfigName, Json()};
+  }
+
+  absl::StatusOr<FilterConfig> GenerateFilterConfigOverride(
+      upb_strview /*serialized_filter_config*/,
+      upb_arena* /*arena*/) const override {
+    return absl::InvalidArgumentError(
+        "router filter does not support config override");
```

Comment:
Should this be documented in the spec https://github.com/grpc/proposal/blob/master/A39-xds-http-filters.md#the-router-filter?

Reply:
I don't think it needs to be called out specifically.  The default assumption should be that a filter does not support any override config, unless it is explicitly documented to do so.

Reply:
>The default assumption should be that a filter does not support any override config

Might be better to call out the NACK behavior because the current spec is only saying
```
Note that, for a given request, the filter's configuration will come from the top-level filter list in the 
HttpConnectionManager config and the most specific override. In other words, if a given filter
instance has an override in the ClusterWeight proto, that will be used; otherwise, 
if it has an override in the Route proto, that will be used; otherwise, 
if it has an override in the VirtualHost proto, that will be used; otherwise, no override will be used.
```

This could be misleading because it doesn't say any filter may not support any override config, and should NACK during LDS/RDS update validation time if a filter does not support override config but an override config is present.

Reply:
I've sent you grpc/proposal#226 to clarify this.

Reply:
Thanks for the RP!